aptos-move/move-examples/post_mint_reveal_nft/sources/minting.move

```diff
@@ -0,0 +1,994 @@
+            &resource_signer,
+            source_collection_name,
+            // token name
+            utf8(b""),
+            // token description
+            utf8(b""),
+            // token maximum
+            0,
+            // token uri
+            utf8(b""),
+            signer::address_of(&resource_signer),
+            1,
+            0,
+            create_token_mutability_config(
+                &vector<bool>[ false, false, false, false, true ]
+            ),
+            vector<String>[string::utf8(BURNABLE_BY_OWNER)],
+            vector<vector<u8>>[bcs::to_bytes<bool>(&true)],
+            vector<String>[string::utf8(b"bool")],
+        );
+
+        move_to(&resource_signer, SourceToken { token_data_id: source_token_data_id });
+    }
+
+    /// Set the minting and reveal config of this collection.
+    public entry fun set_minting_and_reveal_config(
+        admin: &signer,
+        whitelist_minting_start_time: u64,
+        whitelist_minting_end_time: u64,
+        whitelist_mint_price: u64,
+        public_minting_start_time: u64,
+        public_minting_end_time: u64,
+        public_mint_price: u64,
+        reveal_time: u64,
+    ) acquires NFTMintConfig, WhitelistMintConfig, PublicMintConfig, RevealConfig {
+        let nft_mint_config = borrow_global_mut<NFTMintConfig>(@post_mint_reveal_nft);
+        assert!(signer::address_of(admin) == nft_mint_config.admin, error::permission_denied(ENOT_AUTHORIZED));
+
+        let now = timestamp::now_seconds();
+        // assert that we are setting the whitelist time to sometime in the future
+        assert!(whitelist_minting_start_time > now && whitelist_minting_start_time < whitelist_minting_end_time, error::invalid_argument(EINVALID_TIME));
+        // assert that the public minting starts after the whitelist minting ends
+        assert!(public_minting_start_time > whitelist_minting_end_time && public_minting_start_time < public_minting_end_time, error::invalid_argument(EINVALID_TIME));
+        // assert that the public minting price is equal or more expensive than the whitelist minting price
+        assert!(public_mint_price >= whitelist_mint_price, error::invalid_argument(EINVALID_PRICE));
+        assert!(reveal_time >= public_minting_end_time, error::invalid_argument(EINVALID_TIME));
+
+        if (exists<WhitelistMintConfig>(@post_mint_reveal_nft)) {
+            let whitelist_mint_config = borrow_global_mut<WhitelistMintConfig>(@post_mint_reveal_nft);
+            whitelist_mint_config.whitelist_minting_start_time = whitelist_minting_start_time;
+            whitelist_mint_config.whitelist_minting_end_time = whitelist_minting_end_time;
+            whitelist_mint_config.whitelist_mint_price = whitelist_mint_price;
+        } else {
+            let resource_account = create_signer_with_capability(&nft_mint_config.signer_cap);
+            move_to(&resource_account, WhitelistMintConfig {
+                whitelisted_address: bucket_table::new<address, u64>(128),
```

Comment:
nit: Can you add a comment explaining why 128?

Reply:
just think 128 is a reasonable number to start with given the size of most collections, but can use something like 512 / 1024 as well - what do you think?

Reply:
just added more comments

Reply:
I'd suggest making this a constant with very clear documentation in case people want to understand and tweak this.

Reply:
let's use 0 here. It is not easy to justify why 128 makes sense here. I'll start to optimize this to be `SmartTable` soon that will have similar constructor as `SmartVector` so then we can update this file too.